com.github.sdbg.debug.ui/src/com/github/sdbg/debug/ui/internal/chrome/ChromeLaunchShortcut.java

```diff
@@ -69,6 +69,10 @@ protected ILaunchConfigurationType getConfigurationType() {
 
   @Override
   protected void launch(IResource resource, String mode) {
+    launch(resource, mode, null);
+  }
+
+  public void launch(IResource resource, String mode, String url) {
```

Comment:
ChromeLaunchShortCut.java is a code inherited from Dart and not (currently) used in SDBG.

Basically, it implements a standard Eclipse ILaunchShortcut extension point. From a user perspective, this allows you  to right-click an HTML file and launch a DART debugging session for it ("Debug as -> DART application"). For launching, the specially crafted Chrome which has the DartVM is used but that's a detail of course.

Now, we want the same behavior for GWT, but the GWT plugin itself already has the GWT "Web Application" ILaunchShortcut implementation(s), that launch the GWT SDM server. I guess you want to call into SDBG *from there* (or from a new GWT Web App launcher implementation).

Wit the above said, I don't think we want to base ChromeLaunchShortcutExt on top of the dead ChromeLaunchShortcut, which - even worse - implements the ILaunchShortcut API which has no notion of "URL". 

How about:
- Renaming IChromeLaunchShortcutExt to just IChromeLaunch?
- Renaming ChromeLaunchShortcutExt to ChromeLaunchImpl and no longer inheriting from ChromeLaunchShortcut?
- Copying the useful bits and pieces from ChromeLaunchShortcut directly into ChromeLaunchImpl?